ledger/src/blockstore.rs

```diff
@@ -1764,6 +1767,28 @@ impl Blockstore {
         }
     }
 
+    pub fn get_block_height(&self, slot: Slot) -> Result<Option<u64>> {
+        datapoint_info!(
+            "blockstore-rpc-api",
+            ("method", "get_block_height".to_string(), String)
+        );
+        let lowest_cleanup_slot = self.lowest_cleanup_slot.read().unwrap();
+        // lowest_cleanup_slot is the last slot that was not cleaned up by
+        // LedgerCleanupService
+        if *lowest_cleanup_slot > 0 && *lowest_cleanup_slot >= slot {
+            return Err(BlockstoreError::SlotCleanedUp);
+        }
+        self.block_height_cf.get(slot)
+    }
+
+    pub fn cache_block_height(&self, slot: Slot, block_height: u64) -> Result<()> {
+        if self.get_block_height(slot).unwrap_or_default().is_none() {
```

Comment:
I think it's ok to forgo this check here, if we're going to overwrite this it should be the same value, unless we have need to purge a slot, in which case the newer value should take precedent anyways